### PR TITLE
[Enhancement] Support for asterisk operator about jsonpath for parsing avro union type

### PR DESCRIPTION
**Pull request by zaorangyang**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

We use jsonpath to support AVRO union type parsing. This PR introduces a simple asterisk operator semantics.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/exprs/json_functions.cpp [hunk 1]
```diff
@@ -157,11 +157,34 @@ Status JsonFunctions::extract_from_object(simdjson::ondemand::object& obj, const
         // we have to do some special treatment for the second elem of json path.
         // If the key is not found in json object, simdjson::NO_SUCH_FIELD would be returned.
         if (i == 1) {
-            HANDLE_SIMDJSON_ERROR(obj.find_field_unordered(col).get(tvalue),
-                                  fmt::format("unable to find field: {}", col));
+            if (col == "*") {
+                // There should be no jsonpath for this pattern, $.*
```
Comment (rickif) [inline be/src/exprs/json_functions.cpp:161]: `$.*` is a valid jsonpath.

File: be/src/exprs/json_functions.cpp [hunk 2]
```diff
@@ -157,11 +157,34 @@ Status JsonFunctions::extract_from_object(simdjson::ondemand::object& obj, const
...
+                // There are always two patterns.
+                // 1. {"field_name": null}
+                // 2. {"field_name": {"field_type": data}}
+                // For pattern1, we just return null value.
+                // For pattern2, we get the first field of object as next value.
+                if (tvalue.is_null()) {
+                    return Status::NotFound("null value");
+                }
+                if (tvalue.type() != simdjson::ondemand::json_type::object) {
+                    return Status::InvalidArgument(fmt::format("invalid json path: {}", jsonpath[i].key));
```
Comment (rickif) [inline be/src/exprs/json_functions.cpp:178]: I think it's better to report the error as `extracting * from non-object type is not supported`, or something else.

File: be/src/exprs/json_functions.cpp (hunk 1 above)
Comment (zaorangyang) [reply be/src/exprs/json_functions.cpp:161]: > `$.*` is a valid jsonpath.

For avro union type, there is always a  field in root object. just like this.
{"field_name": null}
{"field_name": {"field_type": data}}

**Review by rickif — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by chaoyli — APPROVED** [review]

**Pull request merged**